src/pages/family-info/FamilyForm.tsx

```diff
@@ -0,0 +1,170 @@
+	const date = Date.parse(familyMemberData?.birthDate);
+	return !isNaN(date);
+}
+
+export default function FamilyForm(props: FamilyFormProps): React.ReactElement {
+	const { family, onChange, editable } = props;
+
+	const [hasErrorsArray, setHasErrorsArray] = useState<boolean[]>(Array(family.length).fill(false));
+	const [familyMemberDatas, setFamilyMemberDatas] = useState<FamilyMemberData[]>(family.map(parseFamilyMemberData));
+	const [familyIndex, setFamilyIndex] = useState(0);
+
+	const handleDefaultsAjv = createAjv({ useDefaults: true });
+
+	const setFamilyIndexFromButton = (event: React.MouseEvent<HTMLElement>, newIndex: number): void => {
+		newIndex !== null && setFamilyIndex(newIndex);
+	};
+
+	function setCurrentData(dataFamilyMember: FamilyMemberData, errors: unknown[] | undefined): void {
+		const canSaveArrayCopy = hasErrorsArray.slice();
+		canSaveArrayCopy[familyIndex] = !errors || errors.length == 0;
+		setHasErrorsArray(canSaveArrayCopy);
+
+		if (!canParseFamilyMember(dataFamilyMember)) return;
+
+		const familyMemberDatasCopy = familyMemberDatas.slice();
+		familyMemberDatasCopy[familyIndex] = dataFamilyMember;
+		setFamilyMemberDatas(familyMemberDatasCopy);
+	}
+
+	function canSave(): boolean {
+		let canSave = true;
+		hasErrorsArray.forEach((element) => {
+			canSave = canSave && element;
+		});
+		return canSave;
+	}
+
+	const toggleButtons = familyMemberDatas.map((step, index) => {
+		let text = "Familiar " + (index + 1).toString();
+		if (step && step.fullName) text = step.fullName;
+		return (
+			<ToggleButton id={"family" + index.toString()} key={index} value={index}>
+				{text}
+			</ToggleButton>
+		);
+	});
+
+	const translator = (id: string, defaultMessage: string | undefined): string => {
+		if (id.includes("ci.error")) return "Se deben ingresar solo los números, sin puntos ni guiones y no puede quedar vacía";
+		return defaultMessage ?? "";
+	};
+
+	return (
+		<div>
+			<div style={{ display: "flex", flexDirection: "row", justifyContent: "flex-end" }}>
+				<ToggleButtonGroup value={familyIndex} exclusive onChange={setFamilyIndexFromButton}>
+					{toggleButtons}
+
+					{familyMemberDatas.length <= 1 && editable ? (
+						<ToggleButton
+							id="addFamilyMember"
+							value={familyMemberDatas.length}
+							onClick={(): void => setFamilyMemberDatas([...familyMemberDatas, {} as FamilyMemberData])}>
+							+
+						</ToggleButton>
+					) : null}
+				</ToggleButtonGroup>
+			</div>
+			<JsonForms
+				i18n={{ translate: translator as Translator }}
+				schema={schema as JsonSchema7}
+				uischema={ui}
+				data={familyMemberDatas[familyIndex]}
+				renderers={materialRenderers}
+				cells={materialCells}
+				onChange={({ data, errors }): void => setCurrentData(data, errors)}
+				readonly={!editable}
+				ajv={handleDefaultsAjv}
+			/>
+			<div style={{ display: "flex", flexDirection: "row", justifyContent: "flex-end" }}>
+				{editable ? (
```

Comment:
generalmente no tengo drama con los styles in lune pero ya tenes el .scss creado y los styles se repiten con la linea 130, se podria poner un className y pasar esas reglas al .scss.

Reply:
ok, me explicas bien como en alguna llamada?